### DISCUSSION TITLE
Permission denied when building application

### DISCUSSION THREAD

**Discussion by Sobch** [vercel/next.js#33092] [Help]
### What version of Next.js are you using?

12.0.7

### What version of Node.js are you using?

15.8.0

### What browser are you using?

none

### What operating system are you using?

FreeBSD 12.2-RELEASE-p5

### How are you deploying your application?

npm run build (next build)

### Describe the Bug

After migrating from version 11.1.2 to 12.0.7, I've started getting permission error when building next app. That's true, I don't have permission to /usr/home, because it's hosting server. I actually have permission to /usr/home/USER.

I've reproduced this issue on new, clean project created using `yarn create next-app --typescript` command. One thing I had to do was to disable SWC (probably not having impact on the issue).

Trying again on version 11.1.2 passed without issues.

```
> next build

info  - Checking validity of types  
warn  - No ESLint configuration detected. Run next lint to begin setup
info  - Disabled SWC as replacement for Babel because of custom Babel configuration ".babelrc" https://nextjs.org/docs/messages/swc-disabled
info  - Using external babel configuration from /usr/home/USER/domains/test_app/.babelrc
info  - Creating an optimized production build ..<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies
info  - Creating an optimized production build ...<w> [webpack.cache.PackFileCacheStrategy] Caching failed for pack: Error: Unable to snapshot resolve dependencies
info  - Creating an optimized production build  
info  - Compiled successfully

> Build error occurred
[Error: EACCES: permission denied, readlink '/usr/home'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'readlink',
  path: '/usr/home'
}
info  - Collecting page data .npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! test_app@ build: `next build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the test_app@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/USER/.npm/_logs/2022-01-07T15_50_10_793Z-debug.log

```

### Expected Behavior

Building project does not require any parent directory access permissions.

### To Reproduce

Probably restricting access to parent directories (/usr/home) and trying to build a project.

**Comment by marcin-stasiak** [comment]


**Reply by kmbt** [reply]
Actually, you just need to enable the permission for execution of files at your account. The option is documented right here:
https://wiki.mydevil.net/Binexec

**Comment by Sobch** [comment]
Yes, it is. I know that hosting is forbidding some rights, but it was strange for me it started to happen after version upgrade.

**Comment by Sobch** [comment]
Discussion is up to be closed, maybe Next.js should not be dependent on rights to directories different than project, but to fix the problem, right now is to keep older version or change hosting. Thanks for your interest.
